src/network/network_gui.cpp

```diff
@@ -1564,333 +1575,738 @@ NetworkCompanyInfo *GetLobbyCompanyInfo(CompanyID company)
+						GuiShowTooltips(this, STR_NETWORK_CLIENT_LIST_PLAYER_ICON_HOST_TOOLTIP, 0, nullptr, close_cond);
+						return true;
+					}
+				}
 
-	uint server_client_width;
-	uint line_height;
+				ButtonCommon *button = this->GetButtonAtPoint(pt);
+				if (button == nullptr) return false;
 
-	Dimension icon_size;
+				GuiShowTooltips(this, button->tooltip, 0, nullptr, close_cond);
+				return true;
+			};
+		}
 
-	NetworkClientListWindow(WindowDesc *desc, WindowNumber window_number) :
-			Window(desc),
-			selected_item(-1)
+		return false;
+	}
+
+	void OnDropdownClose(Point pt, int widget, int index, bool instant_close) override
 	{
-		this->InitNested(window_number);
+		/* If you close the dropdown outside the list, don't take any action. */
+		if (widget == WID_CL_MATRIX) return;
+
+		Window::OnDropdownClose(pt, widget, index, instant_close);
 	}
 
-	/**
-	 * Finds the amount of clients and set the height correct
-	 */
-	bool CheckClientListHeight()
+	void OnDropdownSelect(int widget, int index) override
 	{
-		int num = 0;
+		switch (widget) {
+			case WID_CL_SERVER_VISIBILITY:
+				if (!_network_server) break;
 
-		/* Should be replaced with a loop through all clients */
-		for (const NetworkClientInfo *ci : NetworkClientInfo::Iterate()) {
-			if (ci->client_playas != COMPANY_INACTIVE_CLIENT) num++;
-		}
+				_settings_client.network.server_advertise = (index != 0);
+				break;
+
+			case WID_CL_MATRIX: {
+				StringID text = STR_NULL;
+				QueryCallbackProc *callback = nullptr;
+
+				switch (index) {
+					case -1:
+						/* Return, as we are not going to show a Query window. */
+						return;
+
+					case DD_CLIENT_ADMIN_KICK:
+						_admin_client_id = this->dd_client_id;
+						text = STR_NETWORK_CLIENT_LIST_ASK_CLIENT_KICK;
+						callback = AdminClientKickCallback;
+						SetDParamStr(0, NetworkClientInfo::GetByClientID(_admin_client_id)->client_name);
+						break;
+
+					case DD_CLIENT_ADMIN_BAN:
+						_admin_client_id = this->dd_client_id;
+						text = STR_NETWORK_CLIENT_LIST_ASK_CLIENT_BAN;
+						callback = AdminClientBanCallback;
+						SetDParamStr(0, NetworkClientInfo::GetByClientID(_admin_client_id)->client_name);
+						break;
+
+					case DD_COMPANY_ADMIN_RESET:
+						_admin_company_id = this->dd_company_id;
+						text = STR_NETWORK_CLIENT_LIST_ASK_COMPANY_RESET;
+						callback = AdminCompanyResetCallback;
+						SetDParam(0, _admin_company_id);
+						break;
+
+					case DD_COMPANY_ADMIN_UNLOCK:
+						_admin_company_id = this->dd_company_id;
+						text = STR_NETWORK_CLIENT_LIST_ASK_COMPANY_UNLOCK;
+						callback = AdminCompanyUnlockCallback;
+						SetDParam(0, _admin_company_id);
+						break;
+
+					default:
+						NOT_REACHED();
+				}
+
+				assert(text != STR_NULL);
+				assert(callback != nullptr);
 
-		num *= this->line_height;
+				/* Always ask confirmation for all admin actions. */
+				ShowQuery(
+					STR_NETWORK_CLIENT_LIST_ASK_CAPTION,
+					text,
+					this,
+					callback
+				);
+			} break;
```

Comment:
missing newline

Reply:
Where are you missing a newline?

`break` after `}` is common without newline.
There is a newline after `} break;`

Not sure where you would like one?